compiler/rustc_trait_selection/src/traits/engine.rs

```diff
@@ -202,14 +202,22 @@ where
             .map(|infer_ok| self.register_infer_ok_obligations(infer_ok))
     }
 
+    /// Go over the list of pending obligations and try to evaluate them.
+    ///
+    /// For each result:
+    /// Ok: remove the obligation from the list
+    /// Ambiguous: leave the obligation in the list to be evaluated later
+    /// Err: remove the obligation from the list and return an error
+    ///
+    /// Returns a list of errors from obligations that evaluated to Err.
```

Comment:
inconsistent, why add this comment to `try_evaluate_obligations` and not to `evaluate_obligations_error_on_ambiguity` 

I feel like adding this comment to impls for `TraitEngine` isn't worth it and would only add it here (and maybe the `trait TraitEngine` definition)